account_reconciliation_widget/models/reconciliation_widget.py

```diff
@@ -1020,11 +1020,14 @@ def _get_statement_line(self, st_line):
         statement_currency = (
             st_line.journal_id.currency_id or st_line.journal_id.company_id.currency_id
         )
-        if st_line.amount_currency and st_line.currency_id:
-            amount = st_line.amount_currency
-            amount_currency = st_line.amount
+        st_line_currency = (
+            st_line.foreign_currency_id or st_line.currency_id or statement_currency
```

Comment:
Why `st_line.currency_id` would be different from `statement_currency`?

Reply:
In older versions of Odoo (e.g. `13.0`), `st_line.currency_id` is actually used as to allow different currency for individual bank statement line!  `st_line.foreign_currency_id` does not exists.  So it maybe possible that from migration `st_line.currency_id` is different from `statement_currency`.

**However** from `14.0` onwards, `st_line.foreign_currency_id` is introduced and `st_line.currency_id` is re-defined as "Journal Currency" according to the main code (`14.0/addons/account/models/account_bank_statement.py`).  So `st_line.currency_id` should be identical to `statement_currency`.

I've modified the code to be more resilient against `st_line.currency_id` being mis-used as a foreign currency arising from legacy issues.

I do recommend applying this FIX to `14.0` as well.

Thanks for reviewing!  Btw, I don't know why pre-commit failed...

Reply:
Then, according to your comment, the problem is in the migration of your DB with such mixed data, not something reproducible in current version. You should fix your DB instead of changing the module.

Reply:
Hi, please allow me to clarify.  Above screenshots of the bug and fix are all taken from 15.0 environment.

This is not a DB issue because:
1. The bug is reproducible on fresh demo-data;
2. This is not an isolated case as a similar issue was reported in [CybroOdoo/CybroAddons#246](https://github.com/CybroOdoo/CybroAddons/issues/246) and a similar proposed fix [CybroOdoo/CybroAddons#177](https://github.com/CybroOdoo/CybroAddons/issues/177) -- thanks to [whotopia](https://github.com/whotopia) for letting me know he encountered the similar problem.

I probably confused you when I was trying to explain why `st_line.currency_id` should be same as `statement_currency` and went on to explain why the bug happened.

Anyways let me try again.  This commit aims to add basic support for `foreign_currency_id` so multi-currency bank statements can be re-conciliated with the widget.

**_`st_line.currency_id` should be same as `statement_currency`_**
Below is code from Odoo 15.0's main repository where `st_line.currency_id` is defined, `currency_id` is the Journal Currency which is statement's currency.
![image](https://user-images.githubusercontent.com/69288384/225880640-b818c668-b005-4f89-8516-f02b7f31d0ca.png)


**_What caused this bug exists?_**
This widget code was copied from older Odoo versions.  When Odoo introduced `foreign_currency_id` in 14.0 --- this widget did not get properly updated.

As evident from this image, you can see pre-14.0 versions, `currency_id` is used for handling multi-currency.
![image](https://user-images.githubusercontent.com/69288384/225881592-335ee823-5161-4720-8738-9e65f6a7d904.png)

Reply:
Seems like the problem happened during #437 [15.0][MIG] account_reconciliation_widget:Migration to 15.0:

In #437 [15.0][MIG] account_reconciliation_widget:Migration to 15.0:

The commit "https://github.com/OCA/account-reconcile/pull/437/commits/62cd65ca8fe22670c62b3546c38bdfa270e78a4e Fix foreign_currency_id vs currency_id on statement" actually fixes the issue.

But then the commit is over-written in https://github.com/OCA/account-reconcile/pull/437/commits/7f6d1a2ec6d0361479a53d03676208c197458a9c ... not sure why.

Odoo does use `foreign_currency_id` in `15.0` evidently from:
https://github.com/odoo/odoo/blob/15.0/addons/account/models/account_bank_statement.py see Line 549